libdisco/disco.go

```diff
@@ -127,47 +127,47 @@ type handshakeState struct {
 // Serialize is a helper function to serialize a handshake state, later to be unserialized via
 // the `RecoverState()` function.
 // For security purposes, the long-term static keypair is not serialized. Same for the psk
-func (hs handshakeState) Serialize() []byte {
+func (h *handshakeState) Serialize() []byte {
```

Comment:
can we keep `hs`?

Also for changes like this, it's better to split them in multiple PRs : o

Reply:
I'll update to hs to make it consistent across the API.